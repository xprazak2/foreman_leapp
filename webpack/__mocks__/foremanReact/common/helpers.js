import { snakeCase } from 'lodash';

export const deepPropsToSnakeCase = obj =>
  deepPropsToCase(snakeCase, 'propsToSnakeCase only takes objects')(obj);

const deepPropsToCase = (casingFn, errorMsg) => obj => {
  if (typeof obj !== 'object' || obj === null) {
    return obj;
  }
  if (Array.isArray(obj)) {
    return obj.map(deepPropsToCase(casingFn, errorMsg));
  }
  const transformed = propsToCase(casingFn, errorMsg, obj);
  return Object.keys(transformed).reduce((memo, key) => {
    memo[key] = deepPropsToCase(casingFn, errorMsg)(transformed[key]);
    return memo;
  }, {});
};

const propsToCase = (casingFn, errorMsg, ob) => {
  if (typeof ob !== 'object') throw Error(errorMsg);

  return Object.keys(ob).reduce((memo, key) => {
    memo[casingFn(key)] = ob[key];
    return memo;
  }, {});
};
