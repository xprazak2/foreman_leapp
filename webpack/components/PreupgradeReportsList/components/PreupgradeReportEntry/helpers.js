import React from 'react';

import EmptyInfoItem from './EmptyInfoItem';
import StringInfoItem from './StringInfoItem';

export const itemIteratorId = (entry, ...rest) =>
  `${entry.id}-${entry.preupgradeReportId}-${rest.join('-')}`;

export const additionalInfo = entry => {
  const infoAttrs = ['title', 'severity'];

  return infoAttrs.map(attr => {
    const key = itemIteratorId(entry, attr);

    if (!entry[attr]) {
      return <EmptyInfoItem entry={entry} attr={attr} key={key} />;
    }

    switch (attr) {
      case 'title':
        return <StringInfoItem entry={entry} attr={attr} key={key} />;
      case 'severity':
        return <StringInfoItem entry={entry} attr={attr} key={key} />;
      default:
        return '';
    }
  });
};
