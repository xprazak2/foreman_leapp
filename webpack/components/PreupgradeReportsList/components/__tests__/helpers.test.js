import { testSelectorsSnapshotWithFixtures } from '@theforeman/test';

import { additionalInfo } from '../helpers';

const reportEntry = {
  title: 'This is title',
  severity: 'low',
  id: 5,
  preupgradeReportId: 6,
};

const fixtures = {
  'should return additional info': () => additionalInfo(reportEntry),
};

describe('PreupgradeReportList/helpers', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
