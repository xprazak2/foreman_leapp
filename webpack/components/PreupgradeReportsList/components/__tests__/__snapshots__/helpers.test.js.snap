// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreupgradeReportList/helpers should return additional info 1`] = `
Array [
  <StringInfoItem
    attr="title"
    elipsed={false}
    entry={
      Object {
        "id": 5,
        "preupgradeReportId": 6,
        "severity": "low",
        "title": "This is title",
      }
    }
    mapAttr={[Function]}
    translate={false}
  />,
  <StringInfoItem
    attr="severity"
    elipsed={false}
    entry={
      Object {
        "id": 5,
        "preupgradeReportId": 6,
        "severity": "low",
        "title": "This is title",
      }
    }
    mapAttr={[Function]}
    translate={false}
  />,
]
`;
